ir/src/main/scala/org/scalajs/core/ir/Serializers.scala

```diff
@@ -653,7 +690,29 @@ object Serializers {
         case TagIf       => If(readTree(), readTree(), readTree())(readType())
         case TagWhile    => While(readTree(), readTree(), readOptIdent())
         case TagDoWhile  => DoWhile(readTree(), readTree(), readOptIdent())
-        case TagTry      => Try(readTree(), readIdent(), readTree(), readTree())(readType())
+
+        case TagTry =>
+          if (!useHacks068) {
+            sys.error("Invalid tag TagTry")
+          }
+
+          val block = readTree()
+          val errVar = readIdent()
+          val handler = readTreeAllowEmpty()
+          val finalizer = readTreeAllowEmpty()
+          val tpe = readType()
+
+          val maybeCatch = handler.fold(block)(handler =>
+              TryCatch(block, errVar, handler)(tpe))
+          finalizer.fold(maybeCatch)(finalizer =>
+              TryFinally(maybeCatch, finalizer)(tpe))
```

Comment:
With `()`, the `handler =>` part should be on the next line, since it's one normal argument, that happens to be a lambda. (It's different from the `{}` case.)